src/search.js

```diff
@@ -0,0 +1,56 @@
+import { ToastMark } from '@toast-ui/toastmark';
+
+function getChildNodes(parent) {
+  const nodes = [];
+  let curr = parent.firstChild;
+  while (curr) {
+    nodes.push(curr);
+    curr = curr.next;
+  }
+  return nodes;
+}
+
+/**
+ * @param md a markdown string
+ * @return an array of objects with attributes `name`, `level` corresponding to document headings.
+ * provided in order of occurrence in the document
+ */
+export function searchHeadings(md) {
+  const ast = new ToastMark(md);
+  const headings = []; // this is mutated
+  bfsHeaderAccumulate(ast.getRootNode(), headings);
+  return headings;
+}
+
+/**
+ * accumulate headings within an AST into an array. if `node` is a heading itself, with children, those children are accumulated to a
+ * single string. the input `headings` array is mutated
+ */
+function bfsHeaderAccumulate(node, headings) {
+  if (node.type === 'heading') { // TODO how does this handle different header levels?
```

Comment:
Slacked about this, but do you mean between H2 vs H1 vs Hxx? Because here you are just finding all heading nodes regardless. In either case, doesn't seem to be too hard of a feature addition :)

Reply:
this was an old TODO i forgot to clean up! notice that the output from `searchHeadings` includes a `level` attribute for each heading. at the time of the TODO, I was unsure how to access the level, but it turned out to be trivially easy :)